### PR TITLE
Add deploy to experimental instructions

### PR DESCRIPTION
**Pull request by kahlouie**
## Description

Adds documentation about how to deploy to experimental as well as how to clean up after yourself.

## Setup

n/a

## Code Review Verification Steps

* [x] Request review from a member of a different team.

### CODE REVIEW

File: docs/how-to/deploy-to-experimental.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# How to Deploy to Experimental
+
+Experimental is the MilMove environment that can be used to test out a branch out. This is especially good if you'd like Product, Design, or the client to test out a change before you merge the code in. It's also a good decision to use experimental to test out particularly risky changes.
```
Comment (lynzt) [inline docs/how-to/deploy-to-experimental.md]: "...test out a branch out."   - awk...

"before you merge the code in."  - `before you merge to master` ??

Comment (chrisgilmerproj) [inline docs/how-to/deploy-to-experimental.md]: "risky changes like [changes to containers, app startup, or connection to data stores](https://github.com/transcom/mymove/blame/master/.github/PULL_REQUEST_TEMPLATE.md#L29)"

File: docs/how-to/deploy-to-experimental.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Edit [the config file](https://github.com/transcom/mymove/blob/master/.circleci/config.yml) by replacing all instances of `placeholder_branch_name` with your branch name. You'll also want to uncomment any of these lines (and related ones), like this:
+
+```sh
+# if testing on experimental, you can disable these tests by using the commented block below.
+  filters:
+    branches:
+      ignore: your_branch_name_goes_here
+```
+
+When you push up to Github, it'll trigger the CircleCI workflow to start. You can view it's progress on [CircleCI's UI](https://circleci.com/gh/transcom/workflows/mymove) and clicking on your branch. If it's succeeded, then it should be up on [SM experimental](https://my.experimental.move.mil/), [Office experimental](https://office.experimental.move.mil/), and [TSP experimental](https://tsp.experimental.move.mil/). Best practices mean you should announce deploys to experimental in the `#dp3-engineering` slack channel, so only one person is trying to use it at the same time.
```
Comment (lynzt) [inline docs/how-to/deploy-to-experimental.md]: "You can view it's progress" - change `it's` to `its`

"If it's succeeded" - `If it succeeded` ??

Comment (chrisgilmerproj) [inline docs/how-to/deploy-to-experimental.md]: Announce in the #dp3-experinental-env channel at least 20 minutes before you intend to deploy. Try to get a 👍 from someone who is commonly using experimental. If no one comments in that timeframe feel free to deploy.

Comment (chrisgilmerproj) [reply docs/how-to/deploy-to-experimental.md]: "it'll trigger the CircleCI workflow to star **immediately**". We definitely want people checking in with #dp3-experimental-env prior to pushing their code change.

File: docs/how-to/deploy-to-experimental.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## I've got a server-side feature flag
+
+You'll need to add it to [the container definition](https://github.com/transcom/mymove/blob/master/config/app.container-definition.json) in the `environment` array with the name of the server-side flag and the value you'd like it to be on experimental.
+
+## Don't forget to clean up your branch for your pr review
+
+All of the [how to do it instructions](#How-to-do-it) need to be reverted so all future branches don't get deployed to experimental. Also, be sure to announce in `#dp3-engineering` that experimental is available for someone else to use!
```
Comment (chrisgilmerproj) [inline docs/how-to/deploy-to-experimental.md]: #dp3-experimental-env

File: docs/how-to/deploy-to-experimental.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  filters:
+    branches:
+      ignore: your_branch_name_goes_here
+```
+
+When you push up to Github, it'll trigger the CircleCI workflow to start. You can view it's progress on [CircleCI's UI](https://circleci.com/gh/transcom/workflows/mymove) and clicking on your branch. If it's succeeded, then it should be up on [SM experimental](https://my.experimental.move.mil/), [Office experimental](https://office.experimental.move.mil/), and [TSP experimental](https://tsp.experimental.move.mil/). Best practices mean you should announce deploys to experimental in the `#dp3-engineering` slack channel, so only one person is trying to use it at the same time.
+
+## I've got a server-side feature flag
+
+You'll need to add it to [the container definition](https://github.com/transcom/mymove/blob/master/config/app.container-definition.json) in the `environment` array with the name of the server-side flag and the value you'd like it to be on experimental.
```
Comment (chrisgilmerproj) [inline docs/how-to/deploy-to-experimental.md]: Best practice is to add the mustache syntax in that file and follow up by adding the environment variable here: https://github.com/transcom/mymove/blob/master/config/env/experimental.env

File: docs/how-to/deploy-to-experimental.md (hunk 1 above)
Comment (chrisgilmerproj) [reply docs/how-to/deploy-to-experimental.md]: I should probably also add "secure migrations" to that list.

Comment (lynzt) [reply docs/how-to/deploy-to-experimental.md]: also add dataloads? (I know I deployed to experimental when loading the `400ng items`)

File: docs/how-to/deploy-to-experimental.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  filters:
+    branches:
+      ignore: your_branch_name_goes_here
+```
+
+Best practices mean you should announce deploys to experimental in the `#dp3-experinental-env` slack channel at least 20 minutes before you intend to deploy. Try to get a 👍 from someone who is commonly using experimental. If no one comments in that time-frame feel free to deploy. Only one person can use experimental at the same time. Any deploys will overwrite what is currently on experimental. When you push up to Github, it'll trigger the CircleCI workflow to start immediately. You can view its progress on [CircleCI's UI](https://circleci.com/gh/transcom/workflows/mymove) and clicking on your branch. If it has succeeded, then it should be available immediately on [SM experimental](https://my.experimental.move.mil/), [Office experimental](https://office.experimental.move.mil/), and [TSP experimental](https://tsp.experimental.move.mil/).
+
+## I've got a server-side feature flag
+
+You'll need to add it to [experimental config file](https://github.com/transcom/mymove/blob/master/config/env/experimental.env).
```
Comment (chrisgilmerproj) [inline docs/how-to/deploy-to-experimental.md]: I'm sorry I was confusing. You need to put it both in https://github.com/transcom/mymove/blob/master/config/app.container-definition.json and in this other file. Use mustache syntax in the first and just the value in the `experimental.env` file.

Comment (kahlouie) [reply docs/how-to/deploy-to-experimental.md]: ahhh. got it. will change

**Pull request merged**